### PR TITLE
Make ensureVisible return a Future

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
You should be able to wait for the scrolling to finish.

### CODE REVIEW

**Review by axel-op — APPROVED** [review]
Please just tell me it's the last one

**Comment by creativecreatorormaybenot** [comment]
@axel-op Looks like it. Switching over required some changes as I mentioned previously 😅 

I think that I need just one more touch to `EnsureVisibleState` actually 😦

File: lib/src/widgets/ensure_visible.dart [hunk 1]
```diff
@@ -38,7 +42,32 @@ class EnsureVisible extends StatefulWidget {
...
+  ///
+  ///   * `preciseAlignment: 0` will scroll your widget to the start of your viewport.
+  ///   * `preciseAlignment: 0.5` will scroll your widget to the center of your viewport.
+  ///   * `preciseAlignment: 1` will scroll your widget to the end of your viewport.
+  ///   * `preciseAlignment: null` or just omitting it will let the package handle the alignment
+  ///     and will just scroll your widget into view.
+  Future<void> ensureVisible({
+    Duration duration = const Duration(milliseconds: 100),
+    Curve curve = Curves.ease,
+    double preciseAlignment,
```
Comment (creativecreatorormaybenot) [inline lib/src/widgets/ensure_visible.dart:61]: I only wanted to add `preciseAlignment` initially, but in the process I realized that it is really counterintuitive to specify the `duration` and `curve` for the whole widget, so I also moved those to the function.

**Comment by creativecreatorormaybenot** [comment]
@axel-op Wanna take a look? I am pretty certain that this is it then ☹️

File: lib/src/widgets/ensure_visible.dart [hunk 2]
```diff
@@ -21,7 +21,8 @@ class EnsureVisible extends StatefulWidget {
 
   const EnsureVisible({
     Key key,
-    @required this.child,
+    @required
+        this.child,
```
Comment (creativecreatorormaybenot) [inline lib/src/widgets/ensure_visible.dart:21]: The formatter did this.

**Comment by axel-op** [comment]
Why did you deprecate some functions and still use them?

**Comment by creativecreatorormaybenot** [comment]
@axel-op I did not deprecate a function. I deprecated the class parameters because they should be passed as function parameters. In order for this to not break existing code, I need to check them.

I do not know good practices for deprecating parameters. If you know them, would be great if you could adjust it 👍 

(`package_analysis` is failing again)

**Comment by axel-op** [comment]
@creativecreatorormaybenot Could you please leave this PR opened for now? I want to check your changes but don't have time for now

**Comment by creativecreatorormaybenot** [comment]
Sure, I am fine using commits as dependencies anyways while the package is not merged to Pub.

**Comment by axel-op** [comment]
> (package_analysis is failing again)

This is because of another package to interact with GitHub I'm working on also 😓
(Edit: my IDE didn't commit all the files I wanted to, so this is fixed…)

**Comment by axel-op** [comment]
@creativecreatorormaybenot I think this should be OK

**Comment by creativecreatorormaybenot** [comment]
Thanks!

**Pull request merged**